plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -0,0 +1,1180 @@
+                                                    UIManager:sendEvent(Event:new("Close"))
+                                                end
+                                            }
+                                            if tweaked_button_count == 1 then
+                                                early_break = true
+                                                break
+                                            end
+                                            tweaked_button_count = tweaked_button_count + 1
+                                        elseif buttons[j][k].id == "search" and not buttons[j][k].enabled then
+                                            buttons[j][k] = {
+                                                id = "forgot",
+                                                text = _("Forgot"),
+                                                callback = function()
+                                                    self.builder_widget:forgotFromDict(item.word)
+                                                    UIManager:sendEvent(Event:new("Close"))
+                                                end
+                                            }
+                                            if tweaked_button_count == 1 then
+                                                early_break = true
+                                                break
+                                            end
+                                            tweaked_button_count = tweaked_button_count + 1
+                                        end
+                                    end
+                                    if early_break then break end
+                                end
+                            end
+                        end
+
+                        self.builder_widget.current_lookup_word = item.word
+                        self.ui:handleEvent(Event:new("LookupWord", item.word, true, nil, nil, nil, tweak_buttons_func))
+                    end
+                })
+            end
+
+            self.builder_widget = VocabularyBuilderWidget:new{
+                title = _("Vocabulary Builder"),
+                item_table = vocab_items,
+                select_items_callback = function(items, start_idx, end_idx)
+                    DB:select_items(items, start_idx, end_idx)
+                end
+            }
+
+            UIManager:show(self.builder_widget)
+        end
+    }
+end
+
+-- Event sent by readerdictionary "LookupWordWithBookTitle"
+function VocabBuilder:onLookupWordWithBookTitle(word, title)
+    if not settings.enabled then return end
+    if self.builder_widget and self.builder_widget.current_lookup_word == word then return true end
+
+    DB:insertOrUpdate({
+        book_title = title,
+        time = os.time(),
+        word = word
+    })
+    return true
+end
+
+return VocabBuilder
```

Comment:
We like to have a newline at the end in case of potential diffs. Clearly nothing will be added below here, but it'd just give a bad example otherwise. ^_^
```suggestion
return VocabBuilder

```